diff --git a/mm/damon/core.c b/mm/damon/core.c
index 06756ed68..cde856f0a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -903,7 +903,8 @@ static void damos_adjust_quota(struct damon_ctx *c, struct damos *s)
 		return;
 
 	/* New charge window starts */
-	if (time_after_eq(jiffies, quota->charged_from +
+	if (!time_in_range_open(jiffies, quota->charged_from,
+				quota->charged_from +
 				msecs_to_jiffies(quota->reset_interval))) {
 		if (quota->esz && quota->charged_sz >= quota->esz)
 			s->stat.qt_exceeds++;
